File: api/src/models/event.ts
import { Schema, model, Types, Document } from 'mongoose';

export interface IEvent extends Document {
  title: string;
  description?: string;
  location: string;
  time: Date;
  tags: string[];
  attendees: Types.ObjectId[];
  imageUrl?: string;
  latitude?: number;   // Added latitude
  longitude?: number;  // Added longitude
}

const eventSchema = new Schema<IEvent>(
  {
    title: { type: String, required: true },
    description: { type: String },
    location: { type: String, required: true },
    time: { type: Date, required: true },
    tags: { type: [String], default: [] },
    attendees: [{ type: Schema.Types.ObjectId, ref: 'User', index: true }],
    imageUrl: { type: String },
    latitude: { type: Number },    // New field
    longitude: { type: Number },   // New field
  },
  { timestamps: true }
);

// Optional: create 2dsphere index if you want geo queries
eventSchema.index({ latitude: 1, longitude: 1 });

export default model<IEvent>('Event', eventSchema);

File: event_buddy_finder/lib/futures/events/domain/entities/event.dart
class EventEntity {
  final String id;
  final String title;
  final String? description;
  final String location;
  final DateTime time;
  final List<String> tags;
  final List<String> attendeesIds;
  final String? imageUrl;
  final double? latitude;
  final double? longitude;

  EventEntity({
    required this.id,
    required this.title,
    this.description,
    required this.location,
    required this.time,
    required this.tags,
    required this.attendeesIds,
    this.imageUrl,
    this.latitude,
    this.longitude,
  });
}
